WordPress/Sniffs/WP/DeprecatedClassesSniff.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Restricts the use of deprecated WordPress classes and suggests alternatives.
+ *
+ * @link    https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#database-queries
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.12.0
+ */
+class WordPress_Sniffs_WP_DeprecatedClassesSniff extends WordPress_AbstractClassRestrictionsSniff {
+
+	/**
+	 * Minimum WordPress version.
+	 *
+	 * This sniff will throw an error when usage of a deprecated class is
+	 * detected if the class was deprecated before the minimum supported
+	 * WP version; a warning otherwise.
+	 * By default, it is set to presume that a project will support the current
+	 * WP version and up to three releases before.
+	 * This variable allows changing the minimum supported WP version used by
+	 * this sniff by setting a property in a custom phpcs.xml ruleset.
+	 *
+	 * Example usage:
+	 * <rule ref="WordPress.WP.DeprecatedClasses">
+	 *  <properties>
+	 *   <property name="minimum_supported_version" value="4.3"/>
+	 *  </properties>
+	 * </rule>
+	 *
+	 * @var string WordPress versions.
+	 */
+	public $minimum_supported_version = '4.5';
+
+	/**
+	 * List of deprecated classes with alternative when available.
+	 *
+	 * To be updated after every major release.
+	 *
+	 * Version numbers should be fully qualified.
+	 *
+	 * @var array
+	 */
+	private $deprecated_classes = array(
+
+		// WP 3.1.0.
+		'WP_User_Search' => array(
+			'alt'     => 'WP_User_Query',
+			'version' => '3.1.0',
+		),
+	);
+
+
+	/**
+	 * Groups of classes to restrict.
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		// Make sure all array keys are lowercase.
+		$keys = array_keys( $this->deprecated_classes );
+		$keys = array_map( 'strtolower', $keys );
+		$this->deprecated_classes = array_combine( $keys, $this->deprecated_classes );
+
+		return array(
+			'deprecated_classes' => array(
+				'classes' => $keys,
+			),
+		);
+
+	} // End getGroups().
```

Comment:
Only the end comments for control structures. The other end comments (class/function) were never being sniffed for. Left these in to be consistent with most other sniffs.